pandas/tests/test_algos.py

```diff
@@ -570,19 +573,32 @@ def test_object_refcount_bug(self):
         for i in range(1000):
             len(algos.unique(lst))
 
+    def test_index_returned(self, index):
+        # GH#57043
+        index = index.repeat(2)
+        result = algos.unique(index)
+
+        # dict.fromkeys preserves the order
+        unique_values = list(dict.fromkeys(index.values))
+        if isinstance(index, MultiIndex):
+            expected = MultiIndex.from_tuples(unique_values, names=index.names)
+        else:
+            expected = Index(unique_values, dtype=index.dtype)
+            if isinstance(index.dtype, DatetimeTZDtype):
+                expected = expected.normalize()
+        tm.assert_index_equal(result, expected, exact=True)
```

Comment:
nit: I think if you make L578-579 above:

```
result = algos.unique(index.repeat(2))
```

then this can just be `expected = index`. Can you check?

Reply:
Unfortunately, there are several indices having repeat values in the fixture, such as these:
```py
    "bool-object": Index([True, False] * 5, dtype=object),
    "bool-dtype": Index([True, False] * 5, dtype=bool),
```

Reply:
Ah, thanks!